plugins/aws/resoto_plugin_aws/resource/cloudwatch.py

```diff
@@ -0,0 +1,136 @@
+        "cloudwatch_period": S("Period"),
+        "cloudwatch_unit": S("Unit"),
+        "cloudwatch_evaluation_periods": S("EvaluationPeriods"),
+        "cloudwatch_datapoints_to_alarm": S("DatapointsToAlarm"),
+        "cloudwatch_threshold": S("Threshold"),
+        "cloudwatch_comparison_operator": S("ComparisonOperator"),
+        "cloudwatch_treat_missing_data": S("TreatMissingData"),
+        "cloudwatch_evaluate_low_sample_count_percentile": S("EvaluateLowSampleCountPercentile"),
+        "cloudwatch_metrics": S("Metrics", default=[]) >> ForallBend(AwsCloudwatchMetricDataQuery.mapping),
+        "cloudwatch_threshold_metric_id": S("ThresholdMetricId"),
+    }
+    arn: Optional[str] = field(default=None)
+    cloudwatch_alarm_description: Optional[str] = field(default=None)
+    cloudwatch_actions_enabled: Optional[bool] = field(default=None)
+    cloudwatch_ok_actions: List[str] = field(factory=list)
+    cloudwatch_alarm_actions: List[str] = field(factory=list)
+    cloudwatch_insufficient_data_actions: List[str] = field(factory=list)
+    cloudwatch_state_value: Optional[str] = field(default=None)
+    cloudwatch_state_reason: Optional[str] = field(default=None)
+    cloudwatch_state_reason_data: Optional[str] = field(default=None)
+    cloudwatch_state_updated_timestamp: Optional[datetime] = field(default=None)
+    cloudwatch_metric_name: Optional[str] = field(default=None)
+    cloudwatch_namespace: Optional[str] = field(default=None)
+    cloudwatch_statistic: Optional[str] = field(default=None)
+    cloudwatch_extended_statistic: Optional[str] = field(default=None)
+    cloudwatch_dimensions: List[AwsCloudwatchDimension] = field(factory=list)
+    cloudwatch_period: Optional[int] = field(default=None)
+    cloudwatch_unit: Optional[str] = field(default=None)
+    cloudwatch_evaluation_periods: Optional[int] = field(default=None)
+    cloudwatch_datapoints_to_alarm: Optional[int] = field(default=None)
+    cloudwatch_threshold: Optional[float] = field(default=None)
+    cloudwatch_comparison_operator: Optional[str] = field(default=None)
+    cloudwatch_treat_missing_data: Optional[str] = field(default=None)
+    cloudwatch_evaluate_low_sample_count_percentile: Optional[str] = field(default=None)
+    cloudwatch_metrics: List[AwsCloudwatchMetricDataQuery] = field(factory=list)
+    cloudwatch_threshold_metric_id: Optional[str] = field(default=None)
+
+    def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
+        super().connect_in_graph(builder, source)
+        for dimension in self.cloudwatch_dimensions:
+            instance_id = dimension.value
+            instance = builder.graph.search_first_all({"kind": "aws_ec2_instance", "id": instance_id})
+            builder.add_edge(from_node=instance, edge_type=EdgeType.default, to_node=self)
+            builder.add_edge(from_node=instance, edge_type=EdgeType.delete, to_node=self)
```

Comment:
```suggestion
            builder.dependant_node(self, reverse=True, delete_reverse=True, clazz=AwsEc2Instance, id=dimension.value)
```
The `add_edge` method works like this: a reference to one end of the edge, a direction and then keyword args that selects the other end. 

If it is connected for default and delete dependencies, I created the `dependant_node` method.
I guess the reverse and reverse_delete is not easy to use - something to improve I guess.
But logically, this should actuallu do what you want.